### PR TITLE
src: fix memory leaking gchar's returned by g_data_input_stream_read_line()

### PR DESCRIPTION
**Pull request by ejoerns**
Signed-off-by: Enrico Jorns <ejo@pengutronix.de>

### CODE REVIEW

File: src/context.c [hunk 1]
```diff
@@ -145,6 +145,8 @@ static gboolean launch_and_wait_variables_handler(gchar *handler_name, GHashTabl
 			g_hash_table_insert(variables, g_strdup(split[0]), g_strdup(split[1]));
 			g_strfreev(split);
 		}
+
+		g_free(outline);
```
Comment (OnkelUlla) [inline src/context.c]: Please additionally fix the memory leaks that occur up in [line 143's](https://github.com/rauc/rauc/blob/c441d4d2001c1413ec570bc3dc331ff85e66c23f/src/context.c#L143) `continue` statement: before continuing the do-loop a `g_strfreev(split)` and a `g_free(outline)` are missing.

File: src/install.c [hunk 2]
```diff
@@ -708,8 +710,10 @@ static gboolean run_bundle_hook(RaucManifest *manifest, gchar* bundledir, const
 		outline = g_data_input_stream_read_line(datainstream, NULL, NULL, NULL);
 		if (outline) {
 			g_clear_pointer(&hookreturnmsg, g_free);
-			hookreturnmsg = outline;
+			hookreturnmsg = g_strdup(outline);
 		}
+
+		g_free(outline);
```
Comment (OnkelUlla) [inline src/install.c]: What does in comparison to the original code the combination of `g_strdup(outline)` in line 713 and `g_free(outline)` in line 716 change other than adding an additional copy operation?

Comment (ejoerns) [reply src/install.c]: You're right. I've first 'fixed' the gfree and then noticed we need to keep at least the latest content of outline. But did not notice this is handled by the freeing `hookreturnmsg` already.

File: src/context.c (hunk 1 above)
Comment (ejoerns) [reply src/context.c]: I've added another commit to also fix this by using auto ptr.

**Pull request merged**